File: Project Euler #10.py
import time
import math


def prime_finder(number):
    prime = [True] * (number + 1)
    for y in range(3, int(math.sqrt(number + 1)), 2):
        if prime[y]:
            prime_list.append(y)
            for x in range(y * y, number + 1, y + y):
                prime[x] = False
    for x in range(y + 2, number + 1, 2):
        if prime[x]:
            prime_list.append(x)
    return sum(prime_list)


if __name__ == "__main__":
    prime_list = [2]
    start = time.time()
    print(prime_finder(1999999))
    end = time.time()
    print(end - start)

File: Project Euler #10.cpp
#include <iostream>
#include <cmath>
#include <vector>
#include <ctime>
#include <numeric>

long long prime_finder(int number) {
    std::vector<int> primelist = { 2 };
    int* prime = new int[number + 1] {};
    std::fill_n(prime, number + 1, 1);
    for (int y = 3; y < std::sqrt(number + 1); y += 2) {
        if (prime[y] == 1) {
            for (int x = y * y; x < number + 1; x += y + y) {
                prime[x] = 0;
            }
        }
    }
    for (int x = 3; x < number + 1; x += 2) {
        if (prime[x] == true) {
            primelist.push_back(x);
        }
    }
    long long sum = std::accumulate(primelist.begin(), primelist.end(), 0LL);
    return sum;}


int main()
{
    int upper_bound = 1999999;
    int start_s = clock();
    long long sum = prime_finder(upper_bound);
    std::cout << sum << std::endl;
    int stop_s = clock();
    std::cout << (stop_s - start_s) / double(CLOCKS_PER_SEC);
    return 0;
}
